### PR TITLE
Initial restructure PoC

### LINKED ISSUE #1078: Site restructure
Restructure the Besu docs site to separate public and private network content. See the [site structure](https://docs.google.com/spreadsheets/d/1ro4UdBPiw4cEQDE56bctmtzDTesP98qEt82W-KoqDno/edit?usp=sharing). This requires the section index feature in the upgrade Material for MkDocs version (https://github.com/hyperledger/besu-docs/pull/1076 must be merged). Use this PR as a proof of concept.

### LINKED ISSUE #1091: Combine mining information
Combine content from [conceptual mining page](https://besu.hyperledger.org/en/stable/Concepts/Mining/) into [how to configure mining page](https://besu.hyperledger.org/en/stable/HowTo/Configure/Configure-Mining/). Wait till https://github.com/hyperledger/besu-docs/pull/1085 is merged.

### LINKED ISSUE #1092: Combine onchain permissioning content
Combine info from [specify interface version page](https://github.com/hyperledger/besu-docs/pull/1085) and [deploy permissioning management dapp](https://besu.hyperledger.org/en/stable/HowTo/Deploy/Production/) into [onchain permissioning page](https://besu.hyperledger.org/en/stable/HowTo/Limit-Access/Updating-Permission-Lists/). Wait till https://github.com/hyperledger/besu-docs/pull/1085 is merged.

### LINKED ISSUE #1093: Update home page
Move [architecture overview](https://github.com/ConsenSys/doctools.action-builder/blob/main/common/custom_theme/home_with_content.html) to the Besu home page and update content to reflect structure change in #1085

### LINKED ISSUE #1094: Move key management content
Combine info from [use wallets for key management page](https://besu.hyperledger.org/en/stable/HowTo/Send-Transactions/Account-Management/) into [create and send transactions page](https://besu.hyperledger.org/en/stable/HowTo/Send-Transactions/Transactions/). Wait for https://github.com/hyperledger/besu-docs/pull/1085 to be merged.

### LINKED ISSUE #1095: Combine and update bootnode content
Combine info from [configure bootnodes page](https://besu.hyperledger.org/en/stable/HowTo/Deploy/Bootnodes/) into [specify bootnodes page](https://besu.hyperledger.org/en/stable/HowTo/Find-and-Connect/Bootnodes/). Rename to "Configure bootnodes." This page should focus on the private network use case – the Mainnet section can be made into a note. Wait for https://github.com/hyperledger/besu-docs/pull/1085 to be merged.

### LINKED ISSUE #1096: Combine TLS content
Combine info from [conceptual TLS page](https://besu.hyperledger.org/en/stable/Concepts/TLS/) into [client and server TLS page](https://besu.hyperledger.org/en/stable/HowTo/Configure/TLS/Configure-TLS/). Rename page to "Client and server TLS." Wait for https://github.com/hyperledger/besu-docs/pull/1085 to be merged.

### LINKED ISSUE #1097: Combine protocol upgrade content
Combine info from [conceptual protocol upgrade page](https://besu.hyperledger.org/en/stable/Concepts/Protocol-Upgrades/) into [how to upgrade protocol page](https://besu.hyperledger.org/en/stable/HowTo/Upgrade/Upgrade-Protocol/). Wait for https://github.com/hyperledger/besu-docs/pull/1085 to be merged.

### PR DESCRIPTION
**Pull request by alexandratran**
Signed-off-by: Alexandra Tran <alexandra.tran@consensys.net>

## Pull request checklist

Use the following list to make sure your PR fits the Besu documentation quality standard.

### Before creating the pull request

Make sure that:

- [x] [All commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] You've read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] You've [previewed your changes locally](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation).

## Describe the change

- Restructure the entire site to separate public and private network content. Common content is maintained in the public section and linked to from the private section overviews.
- Add all redirects for moved, renamed, or deleted pages.
- Update home page, add architecture to home page, add private and public network section overviews.
- Combine and update:
    - Mining content into a "Configure mining" page.
    - All onchain permissioning content into one page.
    - Key management content into "Send transactions" page.
    - Client and server TLS content into one page.
    - Bootnode content into one page.
    - Protocol upgrade content into one page.

Future PRs will address more content and feature updates and additions.

## Issue fixed

fixes #1078 
fixes #1091
fixes #1092 
fixes #1093 
fixes #1094  
fixes #1095
fixes #1096 
fixes #1097

## Impacted parts

For content changes:

- [x] Documentation content
- [x] Documentation page organization

For tool changes:

- [ ] CircleCI workflow
- [ ] Build and QA tools (for example, lint or vale)
- [ ] MkDocs templates
- [x] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] Read the Docs configuration
- [ ] GitHub integration

## After creating your PR and tests have finished

Make sure that:

- [ ] You've fixed any issues raised by the tests.
- [x] You've [previewed your changes on Read the Docs](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation)
  and added a [preview link](#preview).

## Preview

https://hyperledger-besu--1085.org.readthedocs.build/en/1085/

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -8,46 +8,46 @@ hide:
...
-Besu doesn't support [key management](how-to/Send-Transactions/Account-Management.md) inside the
+Besu doesn't support [key management](how-to/send-transactions.md#use-wallets-for-key-management) inside the
 client. You can use [EthSigner](http://docs.ethsigner.consensys.net/en/latest/) with Besu to access
 your key store and sign transactions.
+
+## Besu architecture
+
+The following diagram outlines the Besu high-level architecture.
+
+![Architecture](images/Architecture.png)
```
Comment (bgravenorst) [inline docs/index.md:50]: This image may require an update. Post-merge architecture is probably different.

Comment (alexandratran) [reply docs/index.md:50]: Raised issue https://github.com/hyperledger/besu-docs/issues/1109

**Comment by NicolasMassart** [comment]
I think the plan should be to merge https://github.com/hyperledger/besu-docs/pull/1113 first then update this PR and so you have everything to push a complete content structure update with home page grid boxes and all the things.

File: docs/public-networks/how-to/send-transactions.md [hunk 2]
```diff
@@ -59,3 +59,18 @@ compares the characteristics of both calls.
 | Does not consume gas                                           | Requires gas                                                                                                                  |
 | Synchronous                                                    | Asynchronous                                                                                                                  |
 | Returns the value of a contract function available immediately | Returns transaction hash only. A block might not include all possible transactions (for example, if the gas price is too low). |
+
+## Use wallets for key management
+
+Besu doesn't support key management inside the client. Use:
+
+* [EthSigner](http://docs.ethsigner.consensys.net/en/latest/) with Besu to provide access to your
```
Comment (bgravenorst) [inline docs/public-networks/how-to/send-transactions.md:67]: And Web3Signer?

File: docs/public-networks/how-to/troubleshoot/trace-transactions.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+Enable the `TRACE` API using the
+[`--rpc-http-api`](../../reference/cli/options.md#rpc-http-api) or
+[`--rpc-ws-api`](../../reference/cli/options.md#rpc-ws-api) command line options.
+
+The `TRACE` API has two sets of trace calls, [ad-hoc tracing APIs](#ad-hoc-tracing-apis) and
+[transaction-trace filtering APIs](#transaction-trace-filtering-apis).
+
+## Ad-hoc tracing APIs
+
+These APIs allow different diagnostic options when tracing calls or transactions.
```
Comment (bgravenorst) [inline docs/public-networks/how-to/troubleshoot/trace-transactions.md:18]: These APIs allow you to use the `trace`, `vmTrace`, or `stateDiff` diagnostic options when tracing calls or transactions.

File: docs/public-networks/how-to/use-pow/mining.md [hunk 4]
```diff
@@ -2,7 +2,19 @@
 description: Using Hyperledger Besu for PoW CPU mining
 ---
 
-# Mining
+# Configure mining
+
+Hyperledger Besu supports CPU and GPU mining, which are configured using command line options.
+
+GPU mining support testing used [Ethminer](https://github.com/ethereum-mining/ethminer) with the
```
Comment (bgravenorst) [inline docs/public-networks/how-to/use-pow/mining.md:9]: Might need to rephrase this

File: docs/public-networks/reference/api/index.md [hunk 5]
```diff
@@ -6925,144 +4818,131 @@ If you specify:
...
+| `5`        | ETH   | Goerli  | PoA test network using Clique |
+| `2018`     | ETH   | Dev     | PoW development network       |
+| `1`        | ETC   | Classic | Main Ethereum Classic network |
+| `7`        | ETC   | Mordor  | PoW test network              |
+| `6`        | ETC   | Kotti   | PoA test network using Clique |
+| `212`      | ETC   | Astor   | PoW test network              |
+
+!!! note
+
+    For almost all networks network ID and chain ID are the same.
```
Comment (bgravenorst) [inline docs/public-networks/reference/api/index.md:4864]: For almost all networks, network...

**Comment by bgravenorst** [comment]
I only did a first pass checking language etc. I'll do a structural review next.

File: docs/public-networks/how-to/send-transactions.md (hunk 2 above)
Comment (alexandratran) [reply docs/public-networks/how-to/send-transactions.md:67]: As far as I understand, Web3Signer is focused on signing consensus layer data and works for some custom eth1 infrastructure, but doesn't work with Besu by default.

Comment (non-fungible-nelson) [reply docs/public-networks/how-to/send-transactions.md:67]: Depends on the signing scheme but web3signer supports secp256k1 which can be used with Besu.

**Review by non-fungible-nelson — COMMENTED** [review]
LGTM

**Review by non-fungible-nelson — APPROVED** [review]
All looks good! We may have to tweak as we go forward but this is an awesome first step!

**Review by NicolasMassart — COMMENTED** [review]
This PR looks good to me.
I suggest that you start with this refactoring and in another PR we could add the tags (to improve the visibility of the private public split) and icons on the two private/public sections and tags.

**Pull request merged**